lib/modules/platform/gitlab/readme.md

```diff
@@ -41,6 +41,8 @@ Remember to set `platform=gitlab` somewhere in your Renovate config file.
 If you're using a private [GitLab container registry](https://docs.gitlab.com/ee/user/packages/container_registry/), you must:
 
 - Set the `RENOVATE_HOST_RULES` CI variable to `[{"matchHost": "${CI_REGISTRY}","username": "${GITLAB_USER_NAME}","password": "${RENOVATE_TOKEN}", "hostType": "docker"}]`.
+
+  Alternatively you can set the CI variables `DOCKER_REGISTRY_GITLAB_COM_USERNAME=${GITLAB_USER_NAME}` and `DOCKER_REGISTRY_GITLAB_COM_PASSWORD=${RENOVATE_TOKEN}`. (See [issue #17940](https://github.com/renovatebot/renovate/issues/17940))
```

Comment:
those don't work out of the box,you need to enable detect host rules from env

https://docs.renovatebot.com/self-hosted-configuration/#detecthostrulesfromenv

Reply:
Added this now, thanks. Not sure about the relative link however.